File: users/allTests/e2eTests/pgroupAddForm.py
# -*- coding: utf-8 -*-

from django.contrib.staticfiles.testing import StaticLiveServerTestCase
from subprocess import call

from users.models import PersonGroup, Person, SysUser
from .userLogin import create_test_user


class PersonGroupAddFormTest(StaticLiveServerTestCase):
    """All tests with nightwatch and Selenium server"""

    def test_pgroupAddForm(self):
        create_test_user()
        self.assertEqual(0,
                         call('nightwatch --test users/allTests/e2eTests/pgroupAddForm.js', shell=True),
                         'Nighwatch tests failed')
        group = PersonGroup.objects.all()[0]
        self.assertEqual(group.name, 'Group name')
        self.assertEqual(group.description, 'This is some group description')

File: users/allTests/e2eTests/pgroupAddForm.js
module.exports = {
    'login to system': function (browser) {
        browser
            .login('test_user', 'test_user')
    },
    'testing person group add form': function (browser) {
        browser
            .url('http://127.0.0.1:8081/#/users')
            .waitForElementVisible('button[ng-click="openGroupAddModal()"]', 2000)
            .click('button[ng-click="openGroupAddModal()"]')
            .waitForElementVisible('form[name=pgroupAddForm]', 2000)
            .setValue('input[id="id_name"]', 'Group name')
            .setValue('textarea[id="id_description"]', 'This is some group description')
            .waitForElementVisible('button[ng-click="ok()"]', 2000)
            .click('button[ng-click="ok()"]')
            .pause(500)
            .end()
    }
};